custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js

```diff
@@ -63,27 +97,38 @@ function MonthModalController($location, $uibModalInstance, dateHelperService) {
     };
 
     vm.onSelectYear = function (item) {
+        vm.months = vm.monthsCopy;
         if (item.id === new Date().getFullYear()) {
-            vm.months = _.filter(vm.monthsCopy, function(month) {
+            vm.months = _.filter(vm.monthsCopy, function (month) {
                 return month.id <= new Date().getMonth() + 1;
             });
 
-            if (startDate === 2019) {
-                vm.months.shift();
-            }
-            
             vm.selectedMonth = vm.selectedMonth <= new Date().getMonth() + 1 ? vm.selectedMonth : new Date().getMonth() + 1;
-        } else if (startDate === 2019 && vm.selectedYear === 2019) {
-            vm.months = _.filter(vm.monthsCopy, function (month) {
-                return month.id >= 2;
-            });
-        } else if (item.id === 2017) {
-            vm.months = _.filter(vm.monthsCopy, function (month) {
-                return month.id >= 3;
-            });
-            vm.selectedMonth = vm.selectedMonth >= 3 ? vm.selectedMonth : 3;
         } else {
-            vm.months = vm.monthsCopy;
+
+            if (isSDD && vm.selectedYear === reportStartDates['sdd'].getFullYear()) {
```

Comment:
I think a bit of refactoring could share a lot of this logic with the version above, no?

Reply:
right. Sure, Lets me do it.